sale_stock_picking_invoicing/wizards/stock_invoice_onshipping.py

```diff
@@ -12,37 +12,88 @@ class StockInvoiceOnshipping(models.TransientModel):
 
     def _build_invoice_values_from_pickings(self, pickings):
         invoice, values = super()._build_invoice_values_from_pickings(pickings)
-        pick = fields.first(pickings)
-        if pick.sale_id:
-            values.update(
-                {
-                    "partner_id": pick.sale_id.partner_invoice_id.id,
+
+        sale_pickings = pickings.filtered(lambda pk: pk.sale_id)
+        # Refund case don't get values from Sale Dict
+        # TODO: Should get any value?
+        if sale_pickings and self._get_invoice_type() != "out_refund":
+            # Case more than one Sale Order the fields below will be join
+            # the others will be overwritting, as done in sale module,
+            # one more field include here Note
+            payment_refs = set()
+            refs = set()
+            # Include Narration
+            narration = set()
+            for pick in sale_pickings:
+                # Other modules can included new fields in Sale Order and include
+                # this fields in the dict of creation Invoice from sale, for
+                # example:
+                # - account_payment_sale
+                #  https://github.com/OCA/bank-payment/blob/14.0/
+                #  account_payment_sale/models/sale_order.py#L41
+                # - sale_commssion
+                #  https://github.com/OCA/commission/blob/14.0/
+                #  sale_commission/models/sale_order.py#L64
+                # To avoid the necessity of a 'glue' module the method get the
+                # values from _prepare_invoice but removed some fields of the
+                # original method, given priority for values from
+                # stock_picking_invoicing dict, for now it's seems the best to
+                # way to avoid the 'glue' modules problem.
+                sale_values = pick.sale_id._prepare_invoice()
+                # Fields to Join
+                # origins.add(sale_values["invoice_origin"])
+                payment_refs.add(sale_values["payment_reference"])
+                refs.add(sale_values["ref"])
+                narration.add(sale_values["narration"])
+
+                # Original dict from sale module, fields uncomment will be remove
+                vals_to_remove = {
+                    # "ref": self.client_order_ref or '',
+                    "move_type",
+                    # "narration": self.note,
+                    "currency_id",
+                    # "campaign_id": self.campaign_id.id,
```

Comment:
As the module is not in a draft state, I think these comments should be removed, maybe concatenate everything in one line as a comment

Reply:
I separated Fields do Get and to Remove, but let as reference, if necessary I can remove it